src/extension.js

```diff
@@ -15,9 +15,11 @@ const { parseAu3CheckOutput } = require('./diagnosticUtils');
 const { config } = require('./ai_config').default;
 
 let diagnosticCollection;
+let consoleOutput;
 
 const checkAutoItCode = document => {
   diagnosticCollection.clear();
```

Comment:
@loganch Wouldn't it make more sense to move this line so that it occurs after the `if` statements that follow it? That would prevent the diagnostics from being cleared unnecessarily. It could be moved into `parseAu3CheckOutput` since it is now only called once.

Reply:
The above should have referenced the ` diagnosticCollection.clear();` line.

Reply:
One minor issue I see is that if there are problems related to an include file, the diagnostics for the include file remain visible even after the main au3 file is closed.

Reply:
Can you elaborate? I don't think I'm seeing that on my end.

Reply:
`Testing.au3`

```autoit
#include  "test2.au3"

test2()

; intention error follows
111
```

`Testing2.au3`
```autoit
func test2()
; intentional error follows
$x = 
endfunc
```

- After opening Testing.au3, the Problem tab looks like this - 

![image](https://user-images.githubusercontent.com/7101313/212521134-fb67a45c-3b5b-42fc-9571-056217edb6ec.png)

- After closing Testing.au3, the Problem tab looks like this - 

![image](https://user-images.githubusercontent.com/7101313/212521174-48ff7fba-499e-4ba3-b75f-0dfac9eeae34.png)

Reply:
Another issue -- If you fix the problem in Test.au3, the `Statement cannot be just an expression.` error shown in the Problem tab doesn't get removed. I'm sure this is because the diagnosticCollection isn't automatically cleared because there is still an error in the included file.

Reply:
Hmmm, okay, I wasn't seeing it because I wasn't introducing syntax errors in included files. 

Apparently AU3Check will just return the line number of the include statement if the error is just "Statement cannot be just an expression."

Reply:
> Another issue -- If you fix the problem in Test.au3, the `Statement cannot be just an expression.` error shown in the Problem tab doesn't get removed. I'm sure this is because the diagnosticCollection isn't automatically cleared because there is still an error in the included file.

Yeah, you're right about that, we're currently only deleting the diagnostics from the collection for the document's URI if the file has 0 errors and 0 warnings.

Reply:
> Apparently AU3Check will just return the line number of the include statement if the error is just "Statement cannot be just an expression."

Hmmm... I'm not seeing that if I understand you correctly.

Reply:
@loganch What do you think about moving the above discussion into a separate issue so that this PR can get merged?

Reply:
Agreed